### PR TITLE
feat: Checkpoints

### PR DESCRIPTION
**Pull request by wuharvey**
Adds a checkpoint near the middle of the course. Tracks the fastest checkpoint time completed during this session, and diffs that time every time the checkpoint is passed.

### CODE REVIEW

File: src/store.ts [hunk 1]
```diff
@@ -18,10 +18,11 @@ const controls = {
   right: false,
 }
 
-export const debug = false as const
+export const debug = true as const
```
Comment (drcmda) [inline src/store.ts:21]: needs to be false eventually 😊

Comment (wuharvey) [reply src/store.ts:21]: oops :)

File: src/ui/Checkpoint.tsx [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import { mutation, useStore } from '../store'
+
+const formatCheckpointDifference = (diffMs: number, negative: boolean) => {
+  return `${!negative ? '+' : ''}${diffMs / 1000}`
+}
+
+export function Checkpoint() {
+  const [showCheckpoint] = useStore((s) => [s.showCheckpoint])
```
Comment (drcmda) [inline src/ui/Checkpoint.tsx:8]: const showCheckpoint = useStore((state) => state.showCheckpoint)

File: src/App.tsx [hunk 3]
```diff
@@ -51,6 +51,24 @@ export function App() {
...
+      mutation.checkpoint1 = mutation.tempCheckpoint1
+      return
+    }
+
+    mutation.checkpointDifference = mutation.tempCheckpoint1 - mutation.checkpoint1
+    if (mutation.checkpointDifference < 0) {
+      mutation.checkpoint1 = mutation.tempCheckpoint1
+    }
+
+    set({
```
Comment (drcmda) [inline src/App.tsx:66]: set({ showCheckpoint: true })

three lines dude 😅

**Review by drcmda — CHANGES_REQUESTED** [review]

File: src/App.tsx (hunk 3 above)
Comment (wuharvey) [reply src/App.tsx:66]: oof, i assumed prettier would handle that 😢

**Pull request merged**